packages/core/admin/admin/src/content-manager/components/Inputs/index.js

```diff
@@ -218,36 +238,40 @@ function Inputs({
   }
 
   return (
-    <GenericInput
-      attribute={fieldSchema}
-      autoComplete="new-password"
-      intlLabel={{ id: label, defaultMessage: label }}
-      // in case the default value of the boolean is null, attribute.default doesn't exist
-      isNullable={inputType === 'bool' && [null, undefined].includes(fieldSchema.default)}
-      description={description ? { id: description, defaultMessage: description } : null}
-      disabled={shouldDisableField}
-      error={error}
-      labelAction={labelAction}
-      contentTypeUID={currentContentTypeLayout.uid}
-      customInputs={{
-        json: InputJSON,
-        uid: InputUID,
-        media: fields.media,
-        wysiwyg: Wysiwyg,
-        ...fields,
-      }}
-      multiple={fieldSchema.multiple || false}
-      name={keys}
-      onChange={onChange}
-      options={options}
-      placeholder={placeholder ? { id: placeholder, defaultMessage: placeholder } : null}
-      required={fieldSchema.required || false}
-      step={step}
-      type={inputType}
-      // validations={validations}
-      value={inputValue}
-      withDefaultValue={false}
-    />
+    <Suspense fallback={<LoadingIndicatorPage />}>
```

Comment:
I've tried that, but it turns out it adds quite a bit of complexity. I have to create an extra component in order to send all the props directly to the input component, otherwise they end up on Suspense. And that extra component means we end up with a new layer of complexity to avoid the bug where the input gets remounted on each parent render

Personally I think it's not a big deal to wrap some extra stuff in suspense, it doesn't affect the behavior of children not using React.lazy. But it's true that the current location isn't good. We render a Suspense component for each attribute, so if I have 4 custom field attributes on a content type, I'll have 4 loaders at the same time.

Instead, I think it would be nice to move the suspense component higher up, maybe at the form level. This way we'd have one single loader for all the React.lazy children. And currently it's only custom fields that are lazy loaded, but we could do that for native fields as well. For example, the rich text editor is pretty big in the bundle, so it would be nice to only load it when it's needed

Reply:
gotcha, ok yeah at this stage it is probably not worth complicating things. I agree it would be better to have the loader at the form level and it would be good to lazy load all the inputs. Maybe this can be an improvement we make between the alpha and beta, wdyt?